stdlib/os/path.pyi

```diff
@@ -77,14 +78,20 @@ else:
         @overload
         def realpath(filename: AnyStr) -> AnyStr: ...
 
-# In reality it returns str for sequences of StrPath and bytes for sequences
-# of BytesPath, but mypy does not accept such a signature.
-def commonpath(paths: Sequence[AnyPath]) -> Any: ...
-
-# NOTE: Empty lists results in '' (str) regardless of contained type.
-# So, fall back to Any
-def commonprefix(m: Sequence[AnyPath]) -> Any: ...
 def lexists(path: AnyPath) -> bool: ...
+@overload
+def commonpath(paths: Sequence[StrPath]) -> str: ...
+@overload
+def commonpath(paths: Sequence[BytesPath]) -> bytes: ...
+
+@overload
+def commonprefix(m: Sequence[StrPath]) -> str: ...
+@overload
+def commonprefix(m: Sequence[BytesPath]) -> bytes: ...
+@overload
+def commonprefix(m: Sequence[List[_T]]) -> Union[List[_T], Literal[""]]: ...
+@overload
+def commonprefix(m: Sequence[Tuple[_T]]) -> Union[Tuple[_T], Literal[""]]: ...
```

Comment:
I don't think this will work, because `""` is not necessarily an iterable of whatever `_T` is. I'll have to come up with something better for this.